frb_dart/lib/src/helpers.dart

```diff
@@ -166,3 +166,12 @@ List<UuidValue> wire2apiUuids(Uint8List raw) {
           Uint8List.view(raw.buffer, i * uuidSizeInBytes, uuidSizeInBytes)),
       growable: false);
 }
+
+List<T?> mapNonNull<T, I>(List<I?> items, T Function(I) mapper) {
+  final out = List<T?>.filled(items.length, null);
+  for (var i = 0; i < items.length; ++i) {
+    final item = items[i];
+    if (item != null) out[i] = mapper(item);
+  }
+  return out;
+}
```

Comment:
Alternatively, we may write `items.map((item) => item == null ? null : mapper(item)).toList()`. I am not sure whether that one will be less performant though (i.e. can Dart compiler optimize it).

Reply:
This might require benchmarking to verify, but map() is known to be slow in transpiled JS (for-loops get JIT'd better) and AOT (indirect calls may not be inlined). Since this is reused many times, I don't mind being a bit more thorough with it.